### DISCUSSION TITLE
I am unable to install mysqlclient in poetry

### DISCUSSION THREAD

**Discussion by AshfaqWanclouds** [python-poetry/poetry#8028] [General]
My toml file is 
****************************************************************************************************************************************************************
[tool.poetry]
name = "vpcplus-gcp-be"
version = "0.1.0"
description = ""
authors = ["AshfaqWanclouds <ashfaq@wanclouds.net>"]
readme = "README.md"
packages = [{include = "vpcplus_gcp_be"}]

[tool.poetry.dependencies]
python = "^3.7"
alembic="^1.7.7"
boto3="^1.21.5"
celery="^5.2.2"
celery-singleton="^0.3.1"
croniter="^1.0.6"
Flask="^2.2.5"
Flask-Compress="^1.8.0"
Flask-Migrate="^2.1.1"
Flask-MySQLdb="^0.2.0"
Flask-Script="^2.0.6"
Flask-SQLAlchemy="^2.5.0"
google-api-core="^2.2.2"
google-api-python-client="^2.29.0"
google-auth="^2.3.3"
grpcio="^1.41.1"
gunicorn="^20.0.4"
kubernetes="^21.7.0"
marshmallow="^3.13.0"
protobuf="^3.19.5"
PyYAML="^6.0"
redis="^4.5.4"
requests="^2.31.0"
SQLAlchemy="^1.4.35"

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
***************************************************************************************************************************************************************

sudo poetry install
***************************************************************************************************************************************************************
Installing dependencies from lock file
Warning: poetry.lock is not consistent with pyproject.toml. You may be getting improper dependencies. Run `poetry lock [--no-update]` to fix it.

Package operations: 13 installs, 0 updates, 0 removals

  • Installing mysqlclient (2.1.1): Failed

  CalledProcessError

  Command '['/root/.cache/pypoetry/virtualenvs/vpcplus-gcp-be-MPmCZpTg-py3.11/bin/python', '-m', 'pip', 'install', '--use-pep517', '--disable-pip-version-check', '--isolated', '--no-input', '--prefix', '/root/.cache/pypoetry/virtualenvs/vpcplus-gcp-be-MPmCZpTg-py3.11', '--no-deps', '/root/.cache/pypoetry/artifacts/a3/e4/9c/a848541c1ee9a5c9c56c24cee94437f951e7f7381d99fb9814e22c19bd/mysqlclient-2.1.1.tar.gz']' returned non-zero exit status 1.

  at /usr/lib/python3.11/subprocess.py:571 in run
       567│             # We don't call process.wait() as .__exit__ does that for us.
       568│             raise
       569│         retcode = process.poll()
       570│         if check and retcode:
    →  571│             raise CalledProcessError(retcode, process.args,
       572│                                      output=stdout, stderr=stderr)
       573│     return CompletedProcess(process.args, retcode, stdout, stderr)
       574│ 
       575│ 

The following error occurred when trying to handle this error:

  EnvCommandError

  Command ['/root/.cache/pypoetry/virtualenvs/vpcplus-gcp-be-MPmCZpTg-py3.11/bin/python', '-m', 'pip', 'install', '--use-pep517', '--disable-pip-version-check', '--isolated', '--no-input', '--prefix', '/root/.cache/pypoetry/virtualenvs/vpcplus-gcp-be-MPmCZpTg-py3.11', '--no-deps', '/root/.cache/pypoetry/artifacts/a3/e4/9c/a848541c1ee9a5c9c56c24cee94437f951e7f7381d99fb9814e22c19bd/mysqlclient-2.1.1.tar.gz'] errored with the following return code 1, and output: 
  Processing /root/.cache/pypoetry/artifacts/a3/e4/9c/a848541c1ee9a5c9c56c24cee94437f951e7f7381d99fb9814e22c19bd/mysqlclient-2.1.1.tar.gz
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
    Getting requirements to build wheel: started
    Getting requirements to build wheel: finished with status 'error'
    error: subprocess-exited-with-error
    
    × Getting requirements to build wheel did not run successfully.
    │ exit code: 1
    ╰─> [32 lines of output]
        /bin/sh: 1: mysql_config: not found
        /bin/sh: 1: mariadb_config: not found
        /bin/sh: 1: mysql_config: not found
        mysql_config --version
        mariadb_config --version
        mysql_config --libs
        Traceback (most recent call last):
          File "/root/.cache/pypoetry/virtualenvs/vpcplus-gcp-be-MPmCZpTg-py3.11/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
            main()
          File "/root/.cache/pypoetry/virtualenvs/vpcplus-gcp-be-MPmCZpTg-py3.11/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
            json_out['return_val'] = hook(**hook_input['kwargs'])
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
          File "/root/.cache/pypoetry/virtualenvs/vpcplus-gcp-be-MPmCZpTg-py3.11/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
            return hook(config_settings)
                   ^^^^^^^^^^^^^^^^^^^^^
          File "/tmp/pip-build-env-5q5hr0o6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel
            return self._get_build_requires(config_settings, requirements=['wheel'])
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
          File "/tmp/pip-build-env-5q5hr0o6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
            self.run_setup()
          File "/tmp/pip-build-env-5q5hr0o6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 488, in run_setup
            self).run_setup(setup_script=setup_script)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
          File "/tmp/pip-build-env-5q5hr0o6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 338, in run_setup
            exec(code, locals())
          File "<string>", line 15, in <module>
          File "/tmp/pip-req-build-uowd6a_o/setup_posix.py", line 70, in get_config
            libs = mysql_config("libs")
                   ^^^^^^^^^^^^^^^^^^^^
          File "/tmp/pip-req-build-uowd6a_o/setup_posix.py", line 31, in mysql_config
            raise OSError("{} not found".format(_mysql_config_path))
        OSError: mysql_config not found
        [end of output]
    
    note: This error originates from a subprocess, and is likely not a problem with pip.
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  

  at /usr/lib/python3/dist-packages/poetry/utils/env.py:1540 in _run
      1536│                 output = subprocess.check_output(
      1537│                     command, stderr=subprocess.STDOUT, env=env, **kwargs
      1538│                 )
      1539│         except CalledProcessError as e:
    → 1540│             raise EnvCommandError(e, input=input_)
      1541│ 
      1542│         return decode(output)
      1543│ 
      1544│     def execute(self, bin: str, *args: str, **kwargs: Any) -> int:

The following error occurred when trying to handle this error:

  PoetryException

  Failed to install /root/.cache/pypoetry/artifacts/a3/e4/9c/a848541c1ee9a5c9c56c24cee94437f951e7f7381d99fb9814e22c19bd/mysqlclient-2.1.1.tar.gz

  at /usr/lib/python3/dist-packages/poetry/utils/pip.py:58 in pip_install
       54│ 
       55│     try:
       56│         return environment.run_pip(*args)
       57│     except EnvCommandError as e:
    →  58│         raise PoetryException(f"Failed to install {path.as_posix()}") from e
       59│

**Comment by dimbleby** [comment]
you almost certainly don't want to `sudo`, you are ignoring the warning about the lockfile being outdated, it looks as though you are using an old poetry... and after all that this is not a poetry question at all - it's a question about how to compile mysqlclient.

https://github.com/PyMySQL/mysqlclient#linux

**Reply by slokhorst** [reply]
To summarize what did it for me on Debian 12:
```bash
apt install python3-dev default-libmysqlclient-dev build-essential
```

Thanks!
